react/features/filmstrip/functions.any.js

```diff
@@ -28,19 +30,33 @@ export function updateRemoteParticipants(store: Object, participantId: ?number)
 
     const {
         fakeParticipants,
+        sortedFakeScreenShareParticipants,
         sortedRemoteParticipants,
         sortedRemoteScreenshares,
         speakersList
     } = state['features/base/participants'];
     const remoteParticipants = new Map(sortedRemoteParticipants);
     const screenShares = new Map(sortedRemoteScreenshares);
+    const screenShareParticipants
+        = sortedFakeScreenShareParticipants ? [ ...sortedFakeScreenShareParticipants.entries() ] : [];
     const sharedVideos = fakeParticipants ? Array.from(fakeParticipants.keys()) : [];
     const speakers = new Map(speakersList);
 
-    for (const screenshare of screenShares.keys()) {
-        remoteParticipants.delete(screenshare);
-        speakers.delete(screenshare);
+    if (getSourceNameSignalingFeatureFlag(state)) {
+        for (const [ ownerId, screenshare ] of screenShareParticipants) {
```

Comment:
if we change the structure of `sortedFakeScreenShareParticipants` to `{participantId: string: displayName: string}` as mentioned above, we'll just iterate through the keys and parse out the ownerId with a utils function